dotcom-rendering/scripts/islands/island-descriptions.mjs

```diff
@@ -0,0 +1,189 @@
+// @ts-check
+
+import { readdir, readFile, writeFile } from 'node:fs/promises';
+import { resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import rehypeAutolinkHeadings from 'rehype-autolink-headings';
+import rehypeSlug from 'rehype-slug';
+import rehypeStringify from 'rehype-stringify';
+import remarkParse from 'remark-parse';
+import remarkRehype from 'remark-rehype';
+import { unified } from 'unified';
+import * as zod from 'zod';
+
+/** @type {(name: string) => RegExp} */
+const getRegExForIsland = (name) =>
+	new RegExp(
+		'\n/\\*\\*\n?' +
+			'((?: \\*.*\n)+)' +
+			' \\*\\/' +
+			`\nexport const ${name} =`,
+		'm',
+	);
+
+const componentsDirectory = resolve(
+	fileURLToPath(import.meta.url),
+	'../../../src/web/components',
+);
+
+const chunk = zod.object({
+	label: zod.string(),
+	isAsset: zod.boolean(),
+	statSize: zod.number(),
+	parsedSize: zod.number(),
+	gzipSize: zod.number(),
+});
+
+/** Sorted by gzipSize */
+const getBundleReport = () =>
+	readFile(
+		resolve(
+			fileURLToPath(import.meta.url),
+			'../../../dist/browser.modern-bundles.json',
+		),
+		'utf-8',
+	)
+		.then((bundle_data) => zod.array(chunk).parse(JSON.parse(bundle_data)))
+		.then((report) => report.sort((a, b) => b.gzipSize - a.gzipSize));
+
+/**
+ * Below 1kB – Good
+ * Below 5kB – Average
+ * Above 5kB – Poor
+ * @type {(gzipSize: number) => '🔴' | '🟠' | '🟢'}
+ */
+const getTrafficLight = (gzipSize) => {
+	if (gzipSize < 1 * 1024) return '🟢';
+	if (gzipSize < 5 * 1024) return '🟠';
+	return '🔴';
+};
+
+const processor = unified()
+	.use(remarkParse)
+	.use(remarkRehype)
+	.use(rehypeSlug)
+	.use(rehypeAutolinkHeadings, { behavior: 'wrap', test: 'h1' })
+	.use(rehypeStringify);
+
+/**
+ * @param {Awaited<ReturnType<typeof getBundleReport>>} report
+ */
+const getIslands = (report) =>
+	readdir(componentsDirectory)
+		.then((components) =>
```

Comment:
more of a style thing, but all these `then`s are hard to follow. you have actually go in and read the code to know what each one does. breaking them down into `const`s with `await`ed values might help that a bit?